src/main/java/org/wildfly/security/audit/SimpleSecurityEventFormatter.java

```diff
@@ -0,0 +1,135 @@
+
+        SecurityIdentity securityIdentity = event.getSecurityIdentity();
+        stringBuilder.append(",security-identity=[name=").append(securityIdentity.getPrincipal().getName());
+        stringBuilder.append(",creation-time=").append(dateFormat.format(Date.from(securityIdentity.getCreationTime()))).append(']');
+
+        return param != null ? null : stringBuilder.append('}').toString();
+    }
+
+    @Override
+    public String handleAuthenticationEvent(SecurityAuthenticationEvent event, StringBuilder param) {
+        checkNotNullParam("event", event);
+        StringBuilder stringBuilder = param != null ? param : new StringBuilder('{');
+
+        handleUnknownEvent(event, stringBuilder);
+        stringBuilder.append(",success=").append(event.isSuccessful());
+
+        return param != null ? null : stringBuilder.append('}').toString();
+    }
+
+    @Override
+    public String handlePermissionCheckEvent(SecurityPermissionCheckEvent event, StringBuilder param) {
+        checkNotNullParam("event", event);
+        StringBuilder stringBuilder = param != null ? param : new StringBuilder('{');
+
+        handleUnknownEvent(event, stringBuilder);
+        stringBuilder.append(",success=").append(event.isSuccessful());
+
+        Permission permission = event.getPermission();
+        stringBuilder.append(",permission=[type=").append(permission.getClass().getName());
+        stringBuilder.append(",actions=").append(permission.getActions());
+        stringBuilder.append(",name=").append(permission.getName()).append(']');
+
+        return param != null ? null : stringBuilder.append('}').toString();
+    }
+
+    /**
+     * Create a new builder.
+     *
+     * @return a new builder.
+     */
+    public static Builder builder() {
+        return new Builder();
+    }
+
+    public static class Builder {
+
+        private Supplier<DateFormat> dateFormatSupplier = SimpleDateFormat::new;
+
+        Builder() {
+        }
+
+        /**
+         * Set a {@link Supplier<DateFormat>} to format any dates in the resulting output.
+         *
+         * @param dateFormatSupplier a {@link Supplier<DateFormat>} to format any dates in the resulting output.
+         * @return {@code this} builder.
+         */
+        public Builder setDateFormatSupplier(Supplier<DateFormat> dateFormatSupplier) {
+            this.dateFormatSupplier = checkNotNullParam("dateFormatSupplier", dateFormatSupplier);
+
+            return this;
+        }
+
+        /**
+         * Build a new {@link SecurityEventVisitor} which will convert {@link SecurityEvent} instances into a simple String
+         *
+         * Once built the Builder can continue to be configured to create additional instances.
+         *
+         * @return a new {@link SecurityEventVisitor} which will convert {@link SecurityEvent} instances into a simple String
+         *         Strings.
+         */
+        public SecurityEventVisitor<?, String> build() {
```

Comment:
Yeah wasn't quite sure on that one - I think I can change this.  The use of a StringBuilder is not public info.